### DISCUSSION TITLE
question: strudel + p5.js

### DISCUSSION THREAD

**Discussion by m-onz** [tidalcycles/strudel#376] [Q&A] [answered]
How might you combine p5.js with this so that strudel is ontop of p5.js?... it would be nice to have audio visual interaction between p5.js and strudel. A very simplistic idea would be to just run a p5.js sketch and have the strudel code ontop but strudel has a canvas built in. Would I need to make that transparent somehow?

Apologies to ask this but I thought I would ask in case someone has already tried it before attempting to hack around with it. I will try to have a go when I have time.

**Comment by felixroos** [comment, marked as answer]
hello! @mikegao45 has started a [discussion](https://github.com/tidalcycles/strudel/discussions/160) about this a while ago. 

> Would I need to make that transparent somehow?

nope, the editor and the canvas are already transparent, the background color comes only from the body. So you can just add a canvas to the body or use the existing one, which you can get with `getDrawContext`, see [example](https://strudel.tidalcycles.org/#Y29uc3QgY3R4ID0gZ2V0RHJhd0NvbnRleHQoKQoKY3R4LmZpbGxTdHlsZT0nYmx1ZScKY3R4LmZpbGxSZWN0KDEwMCwgMTAwLCAxMDAsIDEwMCkKCnNpbGVuY2U%3D). 

There is also a new (and still experimental) [onPaint method](https://strudel.tidalcycles.org?LSRoYxMtA62Q) which helps drawing stuff related to haps (events) within the current time window called `drawTime`, which is currently fixed to [-2,2] but will probably be adjustable in the future. 

let me know if you have any more questions

**Comment by m-onz** [comment]
Nice, I will play around with that and see what I can come up with.
